mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/core/zookeeper/CuratorService.java

```diff
@@ -83,10 +83,13 @@ public void stateChanged(CuratorFramework client, ConnectionState newState) {
     @Override
     public void start() {
         try {
-            if(curator.getState() != CuratorFrameworkState.STARTED) {
-                isConnectedGauge.set(0L);
-                setupCuratorListener();
-                curator.start();
+            synchronized (this) {
```

Comment:
@Andyz26 I think this should handle the potential race when start() can be called more than once (your comment on thread safety).  lmk if you think this is not correct, or can think of a better way...